Thinktecture.Relay.Server/Controller/ManagementWeb/UserController.cs

```diff
@@ -91,7 +106,31 @@ public IHttpActionResult Update(UpdateUser user)
 				return BadRequest();
 			}
 
-			var result = _userRepository.Update(user.Id, user.Password);
+			// OldPassword needs to be correct
+			var authenticatedUser = _userRepository.Authenticate(user.UserName, user.PasswordOld);
+			if (authenticatedUser == null)
+			{
+				return BadRequest("Old password not okay");
+			}
+
+			// new password and repetition need to match
```

Comment:
These checks (and the previous in `Create()` should be moved to a method to reuse them.